packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { readPackageUpSync } from 'read-pkg-up';
+import esbuild from 'esbuild';
+import snakeCase from 'lodash.snakecase';
+import generateESBuildOptions from '../esbuild/generateESBuildOptions.js';
+
+const EXTERNAL_PREFIX = '__holocron_external';
+
+const getExternalLibraryName = (name, version) => [EXTERNAL_PREFIX, snakeCase(name), version.replace(/[^\d.]+/g, '').replace(/\.+/g, '_')].filter(Boolean).join('__');
+
+/**
+ * Transpiles Requires Externals as individual files.
+ * It's completely independent from bundling module's code.
+ */
+export const bundleExternalFallbacks = async () => {
+  const { packageJson } = readPackageUpSync();
+  const { 'one-amex': { bundler = {} } } = packageJson;
+  const { requiredExternals } = bundler;
+
+  if (
+    requiredExternals && Array.isArray(requiredExternals)
+    && requiredExternals.length > 0
+  ) {
+    console.info('Bundling External Fallbacks');
+
+    const {
+      buildExternalsConfig,
+    } = await generateESBuildOptions({ watch: false, useLiveReload: false });
+
+    await Promise.all(['browser', 'node'].map((env) => Promise.all(requiredExternals.map((externalName) => {
+      const indexPath = path.resolve(process.cwd(), 'node_modules', externalName);
+      const buildPath = path.resolve(process.cwd(), 'build', packageJson.version);
+      const externalFilename = `${externalName}.${env}.js`;
+      const outfile = path.resolve(buildPath, externalFilename);
+      const version = readPackageUpSync({
+        cwd: path.resolve(process.cwd(), 'node_modules', externalName),
+      })?.packageJson.version;
+      const envConfig = env === 'browser' ? {
+        globalName: getExternalLibraryName(externalName, version),
+      } : {};
+
+      return esbuild.build({
+        ...buildExternalsConfig(env, externalName),
+        entryPoints: [indexPath],
+        outfile,
+        ...envConfig,
+      }).then(() => {
+        if (env === 'browser') {
+          const content = fs.readFileSync(outfile, 'utf8');
+
+          fs.writeFileSync(
+            outfile,
+            [
+              content,
+              `Holocron.registerExternal({ name: "${externalName}", version: "${version}", module: ${getExternalLibraryName(externalName, version)}});`,
```

Comment:
This step here, of reading the file, injecting some content, then writing it again, looks a _lot_ like an 'injector'

Can an injector be made that is only registered for 'browser' 'externals'? rather than editing this source code outside of esbuilds ecosystem?

By doing this an an injector, this code will still pass through the rest esBuilds pipeline, minifiers, optimizers, syntax-pollyfillers, etc

Reply:
updated to use footer, plugin seems to be a overly complicated solution for appending the build file.
2e54867635687ffe7724456540038955dbd9a1b6

Reply:
Its possibly an overcomplicated solution, but now we have two different places in code that append thing to the end of the built file.

Reply:
Given that this is independent from the module build I don't see this as an issue. It also seems that this is the indented use of the footer api.